cmd/sriov/main.go

```diff
@@ -17,20 +18,50 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
+type envArgs struct {
+	types.CommonArgs
+	MAC types.UnmarshallableString `json:"mac,omitempty"`
+}
+
 func init() {
 	// this ensures that main runs only on main thread (thread group leader).
 	// since namespace ops (unshare, setns) are done for a single thread, we
 	// must ensure that the goroutine does not jump from OS thread to thread
 	runtime.LockOSThread()
 }
 
+func getEnvArgs(envArgsString string) (*envArgs, error) {
+	if envArgsString != "" {
+		e := envArgs{}
+		err := types.LoadArgs(envArgsString, &e)
+		if err != nil {
+			return nil, err
+		}
+		return &e, nil
+	}
+	return nil, nil
+}
+
 func cmdAdd(args *skel.CmdArgs) error {
 	var macAddr string
 	netConf, err := config.LoadConf(args.StdinData)
 	if err != nil {
 		return fmt.Errorf("SRIOV-CNI failed to load netconf: %v", err)
 	}
 
+	envArgs, err := getEnvArgs(args.Args)
+	if err != nil {
+		return fmt.Errorf("SRIOV-CNI failed to parse args: %v", err)
+	}
+
+	if envArgs != nil {
+		hwaddr, err := net.ParseMAC(string(envArgs.MAC))
```

Comment:
Oh, and I also tried to replace it with hardcoded value:
`hwaddr, err := net.ParseMAC("02:02:02:02:02:02")`
Then, the VF interface inside the pod ended up with this MAC (note that it is different from the hardcoded one):
```
72: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq qlen 1000
    link/ether a2:4e:02:22:9f:b5 brd ff:ff:ff:ff:ff:ff
```
On the host level however, the MAC address is updated:
```
6: ens785f3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether fe:ff:e0:1e:3c:04 brd ff:ff:ff:ff:ff:ff
    vf 0 MAC 00:00:00:00:00:00, spoof checking on, link-state auto, trust off
    vf 1 MAC 02:02:02:02:02:02, spoof checking on, link-state auto, trust off
    ...
```
Also because there's a mismatch between the address inside the pod and in the PF configuration, there's no egress until you disable spoof checking as @ahalim-intel has mentioned in the other comment. After you disable spoof checking, source MAC is identical as the one assigned to the interface inside the pod.

This is how it behaves on
```
02:00.3 Ethernet controller [0200]: Intel Corporation Ethernet Controller X710 for 10GbE SFP+ [8086:1572] (rev 01)
02:0e.0 Ethernet controller [0200]: Intel Corporation XL710/X710 Virtual Function [8086:154c] (rev 01)
02:0e.1 Ethernet controller [0200]: Intel Corporation XL710/X710 Virtual Function [8086:154c] (rev 01)
```